Context/Users/Storage/Context/index.ts

```diff
@@ -0,0 +1,31 @@
+import { gracely } from "gracely"
+import { http } from "cloudly-http"
+import { Environment } from "../../../Environment"
+import { Inviter } from "../../../Inviter"
+import { Users } from "./Users"
+
+export class Context {
+	#application: string | undefined
+	get application(): string | undefined {
+		return (this.#application ??= Array.isArray(this.request.header.application)
+			? undefined
+			: this.request.header.application)
+	}
+	#inviter?: Inviter | gracely.Error
+	get inviter(): Inviter | gracely.Error {
+		return (this.#inviter ??= Inviter.open(this.environment, this.application))
+	}
+	#users?: Users | gracely.Error
+	get users() {
+		return (this.#users ??= Users.create(this.state, this))
+	}
+	// public for compatibility with other DO endpoints
```

Comment:
What does this mean?

Reply:
when this started not all endpoints had to change. having public state and environment did not break the old endpoints. in the end they all had to change to some degree and no endpoint is now accessing these public properties. i made them private now.

Reply:
Then we can remove the comment?